siriuspy/siriuspy/devices/scraper.py

```diff
@@ -0,0 +1,283 @@
+"""Storage Ring Scrapers."""
+
+from .device import Device as _Device
+import time as _time
+
+
+class _ScraperDev(_Device):
+    """."""
+
+    class DEVICES:
+        """Devices names."""
+
+        H = 'SI-01SA:DI-ScrapH'
+        V = 'SI-01SA:DI-ScrapV'
+        ALL = (H, V, )
+
+    _properties = (
+        'Home-Cmd', 'ForceComplete-Mon', 'CoordConvErr-Mon',
+        'EnblBacklashComp-Sel', 'EnblBacklashComp-Sts'
+    )
+
+    def __init__(self, devname, properties=None):
+        """."""
+        # check if device exists
+        if devname not in _ScraperDev.DEVICES.ALL:
+            raise NotImplementedError(devname)
+
+        if properties is None:
+            properties = _ScraperDev._properties
+
+        super().__init__(devname, properties)
+
+    @property
+    def is_force_cmd_complete(self):
+        """."""
+        return self['ForceComplete-Mon']
+
+    @property
+    def is_coordinate_convertion_ok(self):
+        """."""
+        return self['CoordConvErr-Mon']
+
+    @property
+    def is_backlash_compensation_enabled(self):
+        """."""
+        return self['EnblBacklashComp-Sts']
+
+    def cmd_enable_backlash_compensation(self):
+        """."""
+        self['EnblBacklashComp-Sel'] = 1
+
+    def cmd_disable_backlash_compensation(self):
+        """."""
+        self['EnblBacklashComp-Sel'] = 0
+
+    def cmd_go_max_aperture(self):
+        """Go to maximum slit aperture."""
+        self['Home-Cmd'] = 1
+
+    def _wait_finish_moving(self, pv_names, timeout=10):
+        """."""
+        t0 = _time.time()
+        ret1 = self._wait(propty=pv_names[0], value=1, timeout=timeout)
+        if not ret1:
+            return ret1
+        timeout -= _time.time() - t0
+        ret2 = self._wait(propty=pv_names[1], value=1, timeout=timeout)
+        return ret2
+
+    @staticmethod
+    def _check_limits(limits):
+        if abs(limits[0]) > abs(limits[1]):
```

Comment:
this won't work every time. I think the following method will:
```python3
    @staticmethod
    def _check_limits(lims, is_negative=False):
        if (lims[1] == lims[0]) or ((lims[1] - lims[0] < 0) ^ is_negative):
            stg = 'larger' if is_negative else 'smaller'
            raise ValueError(f'Inner limit must be {stg:s} than outer limit.')
```

Reply:
Well noted, I implemented your suggestion, thanks!